mshadow/dot_engine-inl.h

```diff
@@ -291,11 +292,48 @@ struct BLASEngine<cpu, float> {
                                   const float *A, int lda, const float *B, int ldb,
                                   float beta, float *C, int ldc, int batch_count,
                                   float **workspace) {
+#if MSHADOW_USE_MKL
```

Comment:
Is cblas_sgemm_batch and cblas_dgemm_batch generally supported in MKL? Do we need to check the version?